app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java

```diff
@@ -1,13 +1,67 @@
 package ar.com.wolox.android.example.ui.login;
 
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
 import javax.inject.Inject;
 
+import ar.com.wolox.android.R;
+import ar.com.wolox.android.example.model.User;
+import ar.com.wolox.android.example.network.LoginService;
+import ar.com.wolox.android.example.network.RetrofitInstance;
 import ar.com.wolox.wolmo.core.presenter.BasePresenter;
-
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class LoginPresenter extends BasePresenter<LoginView> {
 
+
+    @Inject
+    RetrofitServices baseRetrofitServices;
+
     @Inject
     LoginPresenter() {
     }
+
+    void setPreferences(Context context, String emailField) {
+        SharedPreferences sharedPref = context.getSharedPreferences(context.getString(R.string.vlogin_preferences_name), Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = sharedPref.edit();
+        editor.putString("username", emailField);
+        editor.apply();
+
+    }
+
+    void doLogin(String emailField, String passwordField) {
+        /**
+         * emailField = "susan.stevens38@example.com";
+         * String passwordTest = "12345678";
+        */
+        LoginService usersService = RetrofitInstance.getRetrofitInstance().create(LoginService.class);
+        Call<List<User>> call = baseRetrofitServices.getService(LoginService.class).getUserLogin(emailField);
+        call.enqueue(new Callback<List<User>>() {
+
+            @Override
+            public void onResponse(@NotNull Call<List<User>> call, @NotNull Response<List<User>> response) {
+                if (response.body().isEmpty()) {
```

Comment:
you can take out this validations out of onResponse method to be called in another method liek onResponseValidation()